packages/core/test/text/textTests.tsx

```diff
@@ -12,21 +12,46 @@ import * as React from "react";
 import { Classes, Text } from "../../src/index";
 
 describe("<Text>", () => {
-    it("renders jsx text children", () => {
-        const children = (
-            <span>
-                {"computed text "}
+    describe("if ellipsize true", () => {
+        it("truncates string children", () => {
+            const textContent = "textContent";
+            assert.lengthOf(document.getElementsByClassName(Classes.TEXT_OVERFLOW_ELLIPSIS), 0);
+
+            const wrapper = mount(<Text ellipsize>{textContent}</Text>);
+            const element = wrapper.find(`.${Classes.TEXT_OVERFLOW_ELLIPSIS}`);
+            assert.lengthOf(element, 1, `missing ${Classes.TEXT_OVERFLOW_ELLIPSIS}`);
+            assert.strictEqual(element.text(), textContent, "content incorrect value");
+        });
+
+        it("truncates jsx children", () => {
+            const children = (
                 <span>
-                    text in a span
+                    {"computed text "}
+                    <span>
+                        text in a span
+                    </span>
                 </span>
-            </span>
-        );
-        const textContent = "computed text text in a span";
-        assert.lengthOf(document.getElementsByClassName(Classes.TEXT_OVERFLOW_ELLIPSIS), 0);
+            );
+            const textContent = "computed text text in a span";
+            assert.lengthOf(document.getElementsByClassName(Classes.TEXT_OVERFLOW_ELLIPSIS), 0);
+
+            const wrapper = mount(<Text ellipsize>{children}</Text>);
+            const element = wrapper.find(`.${Classes.TEXT_OVERFLOW_ELLIPSIS}`);
+            assert.lengthOf(element, 1, `missing ${Classes.TEXT_OVERFLOW_ELLIPSIS}`);
+            assert.strictEqual(element.text(), textContent, "content incorrect value");
+        });
+    });
+    describe("if ellipsize false", () => {
+        it("doesn't truncate string children", () => {
+            const textContent = "textContent";
+            const className = "bp-test-class";
+            assert.lengthOf(document.getElementsByClassName(Classes.TEXT_OVERFLOW_ELLIPSIS), 0);
```

Comment:
Cool. fwiw I stole the assertion from here: https://github.com/palantir/blueprint/blob/master/packages/core/test/non-ideal-state/nonIdealStateTests.tsx#L16